### PR TITLE
feat: sidebar wallet

### PR DESCRIPTION
**Pull request by lbbest**

### CODE REVIEW

File: client/src/app/_components/ui/mina-button/index.tsx [hunk 1]
```diff
@@ -1,63 +1,87 @@
...
-				{children}
+				{isConnected ? walletDisplayAddress : 'Connect'}
 			</BasicButton>
-			{type == 'chain' ? (
-				<select value={selectedValue} onChange={(e) => handleSelectChange(e)}>
-					<option value="mainnet">mainnet</option>
-					<option value="devnet">devnet</option>
-					<option value="berkeley">berkeley</option>
-				</select>
-			) : null}
```
Comment (ScottJamesPhillips) [inline client/src/app/_components/ui/mina-button/index.tsx:60]: Not sure how you want to play this, my intention was for this component to be flexible in a sense Mina button could be called anywhere and either button could be rendered. Not sure if everywhere we are going to need both the switch and the connect buttons together.
I think previously I was passing in 'connect' or 'chain' depending on which one we wanted rendered.
I think we either stick with that so just put a condition back in rederning whichever button we want, which in my opinion makes it more flexible, or change the component folder name to 'ui/mina-buttons'

File: client/src/app/_components/ui/mina-button/index.tsx [hunk 2]
```diff
@@ -1,63 +1,87 @@
...
+						}}
+					>
+						Switch Chain
+					</BasicButton>
+					<SelectOption
+						id="wallet-chain"
+						name="wallet-chain"
+						value={selectedValue}
+						onChange={(e) => handleSelectChange(e)}
+						options={[
```
Comment (ScottJamesPhillips) [inline client/src/app/_components/ui/mina-button/index.tsx]: Can use the model 'MinaChainDropdownOptions' for these options if possible plz. Not sure I acc did that previously haha

File: client/src/app/_components/ui/mina-button/index.tsx [hunk 3]
```diff
@@ -1,63 +1,87 @@
...
+import { SelectOption } from '../select-option';
 
 export type IMinaButton = {
-	children?: React.ReactNode;
 	disabled?: boolean;
-	checkInstall?: boolean;
-	type?: 'chain' | 'connnect';
 };
 
-export const MinaButton = ({ children, disabled, checkInstall = true, type }: IMinaButton) => {
```
Comment (ScottJamesPhillips) [inline client/src/app/_components/ui/mina-button/index.tsx:19]: For some reason the first callback I created isn't on this version:

const onClickBtn = useCallback(() => {
		if (checkInstall && !window?.mina) {
			alert('No provider was found - please install Auro Wallet');
			return;
		}
		type === 'chain' ? void onClickChain() : void onClickConnect(false);
	}, [checkInstall, onClickChain, onClickConnect, type]);
	
That needs to be added to this component and both buttons/instances need to be triggering it:
	
Screenshot 2024-04-09 at 06 01 19

Will need to keep 'checkInstalled' for that to work, we want that param to display or not display an error to the user telling them about the need for a wallet

**Review by ScottJamesPhillips — COMMENTED** [review]
Have added comments to sections that I think need ammendment

File: client/src/app/_components/ui/mina-button/index.tsx (hunk 1 above)
Comment (lbbest) [reply client/src/app/_components/ui/mina-button/index.tsx:60]: Here I added a prop to the component which accepts an array of types we want rendered. E.g. if we want both button types rendered you would pass in `types={['connect', 'chain']}` or if you only wanted one you would pass in e.g. `types={['connect']}`

File: client/src/app/_components/ui/mina-button/index.tsx (hunk 3 above)
Comment (lbbest) [reply client/src/app/_components/ui/mina-button/index.tsx:19]: Pretty sure I've got this working again but would be great if you could check it out as well

**Comment by ScottJamesPhillips** [comment]
Stuff looks good to me, merge whenever gg

**Pull request merged**